jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/handler/PathMappingsHandlerTest.java

```diff
@@ -0,0 +1,242 @@
+        contextHandler.setContextPath("/");
+
+        PathMappingsHandler pathMappingsHandler = new PathMappingsHandler();
+        contextHandler.setHandler(pathMappingsHandler);
+
+        startServer(contextHandler);
+
+        HttpTester.Response response = executeRequest("""
+            GET / HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+             
+            """);
+        assertEquals(HttpStatus.NOT_FOUND_404, response.getStatus());
+    }
+
+    /**
+     * Test where there are no mappings, and only a wrapper.
+     */
+    @Test
+    public void testOnlyWrapper() throws Exception
+    {
+        ContextHandler contextHandler = new ContextHandler();
+        contextHandler.setContextPath("/");
+
+        PathMappingsHandler pathMappingsHandler = new PathMappingsHandler();
+        pathMappingsHandler.setHandler(new SimpleHandler("WrapperFoo Hit"));
+        contextHandler.setHandler(pathMappingsHandler);
+
+        startServer(contextHandler);
+
+        HttpTester.Response response = executeRequest("""
+            GET / HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+             
+            """);
+        assertEquals(HttpStatus.OK_200, response.getStatus());
+        assertEquals("WrapperFoo Hit", response.getContent());
+    }
+
+    /**
+     * Test where there is only a single mapping, and no wrapper.
+     */
+    @Test
+    public void testOnlyMappingSuffix() throws Exception
+    {
+        ContextHandler contextHandler = new ContextHandler();
+        contextHandler.setContextPath("/");
+
+        PathMappingsHandler pathMappingsHandler = new PathMappingsHandler();
+        pathMappingsHandler.addMapping(new ServletPathSpec("*.php"), new SimpleHandler("PhpExample Hit"));
+        contextHandler.setHandler(pathMappingsHandler);
+
+        startServer(contextHandler);
+
+        HttpTester.Response response = executeRequest("""
+            GET /hello HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+             
+            """);
+        assertEquals(HttpStatus.NOT_FOUND_404, response.getStatus());
+
+        response = executeRequest("""
+            GET /hello.php HTTP/1.1\r
+            Host: local\r
+            Connection: close\r
+             
+            """);
+        assertEquals(HttpStatus.OK_200, response.getStatus());
+        assertEquals("PhpExample Hit", response.getContent());
+    }
+
+    /**
+     * Test where there is only a single mapping, and a wrapper for fallback.
+     */
+    @Test
+    public void testOneMappingAndWrapper() throws Exception
+    {
+        ContextHandler contextHandler = new ContextHandler();
+        contextHandler.setContextPath("/");
+
+        PathMappingsHandler pathMappingsHandler = new PathMappingsHandler();
+        pathMappingsHandler.setHandler(new SimpleHandler("WrapperFoo Hit"));
+        pathMappingsHandler.addMapping(new ServletPathSpec("*.php"), new SimpleHandler("PhpExample Hit"));
```

Comment:
Example 1 (of 2) that might be useful for FCGI (to PHP) Proxy.
This one relies on no-match behavior to fallback to the wrapper.